### PR TITLE
config: warn undefined config item

### PR DESCRIPTION
**Pull request by nolouch**
Signed-off-by: nolouch <nolouch@gmail.com>

### What problem does this PR solve? 
Now, the undefined config item will let PD exit. This makes the new configuration item incompatible with the old configuration template.

### What is changed and how it works?
Only log it with `warn` level.

### Check List 

Tests 
 - Manual test (add detailed scripts or steps below)
 
image

### CODE REVIEW

File: conf/config.toml [hunk 1]
```diff
@@ -83,6 +83,7 @@ max-replicas = 3
 # For example, ["zone", "rack"] means that we should place replicas to
 # different zones first, then to different racks if we don't have enough zones.
 location-labels = []
+# strictly-match-label = false
```
Comment (disksing) [inline conf/config.toml]: Add some comments?

File: conf/config.toml [hunk 2]
```diff
@@ -67,7 +67,7 @@ leader-schedule-limit = 4
 region-schedule-limit = 1024
 replica-schedule-limit = 1024
 merge-schedule-limit = 8
-tolerant-size-ratio = 0
+# tolerant-size-ratio = 0.0
```
Comment (disksing) [inline conf/config.toml]: Why do we need to update this line?

Comment (nolouch) [reply conf/config.toml]:
```
[FATAL] [main.go:54] ["parse cmd flags error"] [error="toml: cannot load TOML value of type int64 into a Go float"
```
The original configuration file cannot be parse.

**Comment by rleungx** [comment]
Personally, I think this change will lose part of its original meaning of checking the config items.

File: conf/config.toml [hunk 3]
```diff
@@ -83,6 +83,8 @@ max-replicas = 3
 # For example, ["zone", "rack"] means that we should place replicas to
 # different zones first, then to different racks if we don't have enough zones.
 location-labels = []
+# Strictly checks if the label of TiKV is matched with LocaltionLabels.
```
Comment (disksing) [inline conf/config.toml]: Location

File: conf/config.toml [hunk 4]
```diff
@@ -83,6 +83,8 @@ max-replicas = 3
 # For example, ["zone", "rack"] means that we should place replicas to
 # different zones first, then to different racks if we don't have enough zones.
 location-labels = []
+# Strictly checks if the label of TiKV is matched with localtion labels.
```
Comment (disksing) [inline conf/config.toml]: location, no `l`.

**Pull request merged**